### PR TITLE
MTDSA-13025: TYS changes for individual-losses-api

### PR DESCRIPTION
**Pull request by harry-ritchie**

### CODE REVIEW

File: app/config/FeatureSwitches.scala [hunk 1]
```diff
@@ -19,17 +19,16 @@ package config
...
 
   def isAmendLossClaimsOrderRouteEnabled: Boolean =
-    maybeConfig match {
-      case Some(configuration) => configuration.getOptional[Boolean]("amend-loss-claim-order.enabled").getOrElse(false)
-      case None                => false
-    }
+      featureSwitchConfig.getOptional[Boolean]("amend-loss-claim-order.enabled").getOrElse(false)
+
+  val isTaxYearSpecificApiEnabled: Boolean = isEnabled("tys-api.enabled")
+  private def isEnabled(key: String): Boolean = featureSwitchConfig.getOptional[Boolean](key).getOrElse(true)
```
Comment (mattstephens-hmrc) [inline app/config/FeatureSwitches.scala:33]: can there be a blank line between lines 32 & 33

File: app/api/models/domain/DownstreamTaxYear.scala [hunk 2]
```diff
@@ -24,11 +25,30 @@ import java.time.format.DateTimeFormatter
   *
   * @param value the tax year string (where 2018 represents 2017-18)
   */
-case class DownstreamTaxYear(value: String) extends AnyVal {
+final case class DownstreamTaxYear private(value: String) {
```
Comment (mattstephens-hmrc) [inline app/api/models/domain/DownstreamTaxYear.scala:28]: needs to be renamed to TaxYear (along with the test). Make sure it's copied from individuals-income-received-api, and the TYS tests are also carried over from TaxYearSpec

File: app/api/endpoints/lossClaim/list/v3/ListLossClaimsController.scala [hunk 3]
```diff
@@ -50,6 +52,7 @@ class ListLossClaimsController @Inject()(val authService: EnrolmentsAuthService,
            businessId: Option[String],
            typeOfClaim: Option[String]): Action[AnyContent] =
     authorisedAction(nino).async { implicit request =>
+      implicit val correlationId: String = idGenerator.getCorrelationId
```
Comment (mattstephens-hmrc) [inline app/api/endpoints/lossClaim/list/v3/ListLossClaimsController.scala:55]: Can this be spaced out a bit, it's all a bit bunched together

File: app/config/FeatureSwitches.scala [hunk 4]
```diff
@@ -19,17 +19,16 @@ package config
 import play.api.Configuration
 import routing.Version
 
-case class FeatureSwitch(maybeConfig: Option[Configuration]) {
+case class FeatureSwitches(featureSwitchConfig: Configuration) {
```
Comment (mattstephens-hmrc) [inline app/config/FeatureSwitches.scala:22]: This also needs the `object FeatureSwitches....` from individuals-income-received-api

Comment (mattstephens-hmrc) [reply app/config/FeatureSwitches.scala:22]: Also FeatureSwitchesSpec needs to be copied over

File: app/config/FeatureSwitches.scala (hunk 1 above)
Comment (harry-ritchie) [reply app/config/FeatureSwitches.scala:34]: Fixed

File: app/api/models/domain/DownstreamTaxYear.scala (hunk 2 above)
Comment (harry-ritchie) [reply app/api/models/domain/DownstreamTaxYear.scala]: Fixed

File: app/api/endpoints/lossClaim/list/v3/ListLossClaimsController.scala (hunk 3 above)
Comment (harry-ritchie) [reply app/api/endpoints/lossClaim/list/v3/ListLossClaimsController.scala:56]: Fixed

File: app/config/FeatureSwitches.scala (hunk 4 above)
Comment (harry-ritchie) [reply app/config/FeatureSwitches.scala:22]: Fixed

**Review by mattstephens-hmrc — APPROVED** [review]

**Pull request merged**